### ISSUE TITLE
strokedPath error

### ISSUE THREAD

**Issue by abu-irrational** [blend2d/blend2d#189]
Here is a BL::Path made of a simple BL::Arc of 120 degrees.
Then, from this Path I build a strokedPath with a given width (thickness)
![immagine](https://github.com/blend2d/blend2d/assets/13416847/8be76c9c-c1e1-473f-b220-6b95af505b28)

As you can see, the stroked-path (p2) has a strange glitch, and it is made of 40 vertices (against 7 vertices of path p1).
Here's a test:
// ------------------------------------------------------------------
BLImage render(const BLContextCreateInfo& cci) {
  BLImage img(256, 256, BL_FORMAT_PRGB32);
  BLContext ctx(img, cci);

  ctx.clearAll();

  BLPath p1;

  double r = 100.0;
  double PI=3.1415;
  
    double r1=r/5*4;
    BLArc xa = BLArc(r,0,r1,r1,PI*2/3,PI*2/3);
    p1.addArc(xa);
 
 	// create a dummyContext
	//  it requires a minimal dummyImage, too
	//  or you won't be able to set options  (--> error INVALID_STATE)
	BLImage dummyImage = BLImage(1,1, BL_FORMAT_A8); 
  BLContext dummyCtx( dummyImage );
  dummyCtx.setStrokeWidth(r/6);

  BLStrokeOptions opt;
  BLPath p2;
  p2.addStrokedPath(p1,dummyCtx.strokeOptions(),blDefaultApproximationOptions);

   ctx.setTransform({1,0,0,1,100,100});
   ctx.strokePath(p2, BLRgba32(0xFFFFFF00));

printf("p1 size is %ld\n", p1.size());
printf("p2 size is %ld\n", p2.size());

  return img;
}

**Comment by kobalicek** [comment]
This is unfortunately a limitation of the current stroker (stroking twice just makes it visible).

This should be resolved by the new stroke, which is not ready atm.

**Comment by kobalicek** [comment]
New fiddle:

```
BLImage render(const BLContextCreateInfo& cci) {
  BLImage img(512, 512, BL_FORMAT_PRGB32);
  BLContext ctx(img, cci);

  BLPath p1;
  double r = 100.0;
  double PI=3.1415;
  double r1=r/5*4;
  BLArc xa = BLArc(r,0,r1,r1,PI*2/3,PI*2/3);
  p1.add_arc(xa);
  bl_debug_path(&p1);

  BLStrokeOptions opt;
  opt.set_width(40.0f);
  BLPath p2;
  p2.add_stroked_path(p1, opt, bl_default_approximation_options);

  ctx.clear_all();
  ctx.set_transform({1,0,0,1,100,100});
  ctx.stroke_path(BLPoint(100, 150), p2, BLRgba32(0xFFFFFF00));

  return img;
}```

**Comment by abu-irrational** [comment]
Petr, I noted you rewrote my old fiddle , translating from the old camelCase notation to the new snake_case notation.

BUT,  the following statement is currently invalid
  opt.set_width(40.0f);    //-->  error: ‘class BLStrokeOptions’ has no member named ‘set_width’

Is it a new method ? 
If so, it would be very convenient, because in my original fiddle I had to use a messy workaround involving a dummyContext.

**Comment by kobalicek** [comment]
Indeed, sorry that doesn't work in the current fiddle. BLStrokeOptions will be a BLObject-like object in the future (and it already is in my local copy). BLStrokeOptions was quirky and had to change.
